### PR TITLE
fix: changes for xserver implementation

### PR DESCRIPTION
**Pull request by robertbolgeo**
# Description

added build files so that the project can be added to the xserver hosting platform.

## Related Issues

Link to any GitHub issues or client requests being addressed in this PR [use"#"](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#referencing-issues-and-pull-requests). If no issue exists, please update [pull-request-board](https://github.com/orgs/pride-land/projects/1).

- Issue/Request: [#230 ](https://github.com/pride-land/frontend-pride-land/issues/230)

## Type of change

Please delete options that are not relevant. (All options may be deleted for minor changes)

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Extensive testing through cURL tests to make sure all expected endpoints and site URLs are working. 
Tested https security by using whynopadlock.com.
Site functionality is continuing to be tested by just clicking and messing around.

- [x] cURL testing: In order to understand front and backend URL connectivity I used cURL testing (Postman would work too) from the command terminal. For anyone looking to duplicate these tests, just choose the endpoint or access point in question and type 'curl https://pridefarm.org/path/to/endpoint/' 
- [x] Feel free to also test the https qualifications of these endpoints by visiting whynopadlock.com and putting the endpoint into the free tester there.

## Screenshots (if applicable)
Including an image of the .htaccess file that is necessary for xserver to serve the static files for the front end.
Screenshot 2024-12-07 at 15 45 07

## Checklist
- [x] I have ensured the code adheres to the coding standards of the project.
- [ ] If changes require documentation updates, I have included them.
- [x] I have performed a self-review of my code.
- [x] New members can easily understand this code, and I have added comments for clarity.
- [x] The PR is client-ready (all client-related feedback has been addressed).

## Additional Context or Notes
Most of the relevant changes didn't actually occur in our code base. Most changes that this PR addresses were done on the xServer server panel. If there are questions on how to use this in the future I think a separate PR request for the necessary documentation for those changes will be necessary.
- This was a pretty hacky fix. There are a lot of separate pieces that could become incompatible if any of the services were to update to stop working together. Some testing to ensure that the fcgi and htaccess files are executable and working correctly would be a good idea. After everything else is smooth I will come back for that.   :clinking_glasses:

### CODE REVIEW

File: lib/cjs/src/Admin/admin-api/admin-layout.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { Image } from "../admin-interface/AdminGalleryTypes";
+export declare const deleteImage: (image: Image) => Promise<any>;
+export declare const getAllImages: () => Promise<any>;
+export declare const getCardRefs: () => Promise<any>;
+export declare const addHeroTagToImg: (image: Image) => Promise<any>;
+export declare const removeHeroTagFromImg: (image: Image) => Promise<any>;
+export declare const getHeroImages: () => Promise<any>;
```
Comment (NabbeunNabi) [inline lib/cjs/src/Admin/admin-api/admin-layout.d.ts:7]: @robertbolgeo Why are these returning `any`?

I am cool with any if the data is controlled by an outside api. But if you know what it returns you should look at response types and maybe make a file that shows them.

Can be addressed in a future ticket. But curious.

File: lib/cjs/src/Admin/admin-api/admin-blogs.d.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export declare const postBlogs: (formData: FormData) => Promise<any>;
```
Comment (NabbeunNabi) [inline lib/cjs/src/Admin/admin-api/admin-blogs.d.ts:1]: ♻️

File: lib/cjs/src/Admin/admin-api/admin-control.d.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export declare const fetchAllStaff: () => Promise<[]>;
```
Comment (NabbeunNabi) [inline lib/cjs/src/Admin/admin-api/admin-control.d.ts]: Does this return an empty promise array?

File: lib/cjs/src/Admin/admin-api/admin-feedbacks.d.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import { FeedbackStatus } from "../admin-interface/AdminFeedbackType";
+export declare const getFeedbacks: () => Promise<any>;
```
Comment (NabbeunNabi) [inline lib/cjs/src/Admin/admin-api/admin-feedbacks.d.ts:2]: ♻️

File: lib/cjs/src/Admin/admin-api/admin-volunteers.d.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import { MinPutVolunteer } from "../admin-interface/AdminVolunteerType";
+export declare const getAllVolunteers: () => Promise<any>;
```
Comment (NabbeunNabi) [inline lib/cjs/src/Admin/admin-api/admin-volunteers.d.ts:2]: ♻️

File: lib/cjs/src/Admin/admin-pages/SideBar.d.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+interface SideBar {
+    elt: any;
```
Comment (NabbeunNabi) [inline lib/cjs/src/Admin/admin-pages/SideBar.d.ts:2]: What is `elt`? I cannot tell by the variable name or the type. Could be something I am not familiar with?

File: lib/cjs/src/Admin/admin-utils/PrivateRoute.d.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import { ReactNode } from 'react';
+declare const PrivateRoute: ({ children }: {
+    children: ReactNode;
+}, { ...rest }: {
+    [x: string]: any;
```
Comment (NabbeunNabi) [inline lib/cjs/src/Admin/admin-utils/PrivateRoute.d.ts:5]: ♻️

File: lib/cjs/src/api/feedbacks.d.ts [hunk 8]
```diff
@@ -0,0 +1,3 @@
+import FeedbackType from "../interfaces/FeedbackType";
+export declare const fetchAllFeedbacks: () => Promise<FeedbackType[] | undefined>;
+export declare const postFeedback: (formData: FormData) => Promise<any>;
```
Comment (NabbeunNabi) [inline lib/cjs/src/api/feedbacks.d.ts:3]: ♻️

File: lib/cjs/src/api/fetchBinaryImage.d.ts [hunk 9]
```diff
@@ -0,0 +1 @@
+export declare const fetchAllBinaryImage: () => Promise<any>;
```
Comment (NabbeunNabi) [inline lib/cjs/src/api/fetchBinaryImage.d.ts:1]: ♻️

File: lib/cjs/src/api/upload-images.d.ts [hunk 10]
```diff
@@ -0,0 +1,2 @@
+export declare const postImage: (formData: FormData) => Promise<any>;
+export declare const fetchAllImages: () => Promise<any>;
```
Comment (NabbeunNabi) [inline lib/cjs/src/api/upload-images.d.ts:2]: ♻️

File: lib/cjs/src/components/blogComponents/blogFocus.d.ts [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import BlogsTypes from "../../interfaces/BlogsType";
+import React, { Dispatch, SetStateAction } from "react";
+interface Props {
+    chosenBlog: BlogsTypes;
+    blogsProp: BlogsTypes[];
+    setChosenBlog: Dispatch<SetStateAction<BlogsTypes | null>>;
```
Comment (NabbeunNabi) [inline lib/cjs/src/components/blogComponents/blogFocus.d.ts:6]: I love this variable naming and typing!!!

File: lib/cjs/src/components/blogComponents/pagination.d.ts [hunk 12]
```diff
@@ -0,0 +1,7 @@
+interface Props {
+    blogsPerPage: number;
+    length: number;
+    handlePagination: any;
```
Comment (NabbeunNabi) [inline lib/cjs/src/components/blogComponents/pagination.d.ts:4]: ♻️

File: lib/cjs/src/components/homeComponents/Section.d.ts [hunk 13]
```diff
@@ -0,0 +1 @@
+export {};
```
Comment (NabbeunNabi) [inline lib/cjs/src/components/homeComponents/Section.d.ts]: Leaving a comment as to why you are exporting an empty object might be helping above this

File: lib/cjs/src/game/EventBus.d.ts [hunk 14]
```diff
@@ -0,0 +1,2 @@
+import { Events } from 'phaser';
+export declare const EventBus: Events.EventEmitter;
```
Comment (NabbeunNabi) [inline lib/cjs/src/game/EventBus.d.ts:2]: 🚌

File: src/Admin/admin-api/admin-layout.tsx [hunk 15]
```diff
@@ -1,59 +1,57 @@
...
+export const deleteImage = async (image: Image) => {
+ const request = await fetch(endpoint + `medias/${image.id}/`, {
+  method: "DELETE",
+  headers: {
+   "Content-Type": "application/json",
+  },
+ });
+ const result = await request.json();
+ return result;
+};
```
Comment (NabbeunNabi) [inline src/Admin/admin-api/admin-layout.tsx:15]: Should use a try catch block to catch any errors and `console.error` so developers can see what the problems are

File: src/Admin/admin-api/admin-layout.tsx [hunk 16]
```diff
@@ -1,59 +1,57 @@
...
+
+export const getAllImages = async () => {
+ const request = await fetch(endpoint + "medias/");
+ const result = await request.json();
+ return result;
+};
```
Comment (NabbeunNabi) [inline src/Admin/admin-api/admin-layout.tsx:21]: ♻️

File: src/Admin/admin-api/admin-layout.tsx [hunk 17]
```diff
@@ -1,59 +1,57 @@
...
+
+export const getCardRefs = async () => {
+ const request = await fetch(endpoint + "cardrefs/");
+ const result = await request.json();
+ return result;
+};
```
Comment (NabbeunNabi) [inline src/Admin/admin-api/admin-layout.tsx:27]: ♻️

File: src/Admin/admin-api/admin-layout.tsx [hunk 18]
```diff
@@ -1,59 +1,57 @@
...
+ const request = await fetch(endpoint + `medias/${image.id}/`, {
+  method: "PATCH",
+  headers: {
+   "Content-Type": "application/json",
+  },
+  body: JSON.stringify({ set_as_hero: true }),
+ });
+ const result = await request.json();
+ return result;
+};
```
Comment (NabbeunNabi) [inline src/Admin/admin-api/admin-layout.tsx:39]: ♻️

File: src/Admin/admin-api/admin-layout.tsx [hunk 19]
```diff
@@ -1,59 +1,57 @@
...
+
+export const removeHeroTagFromImg = async (image: Image) => {
```
Comment (NabbeunNabi) [inline src/Admin/admin-api/admin-layout.tsx:41]: ♻️

File: src/Admin/admin-api/admin-layout.tsx [hunk 20]
```diff
@@ -1,59 +1,57 @@
...
+  headers: {
+   "Content-Type": "application/json",
+  },
+  body: JSON.stringify({ set_as_hero: false }),
+ });
+ const result = await request.json();
+ return result;
+};
+
+export const getHeroImages = async () => {
```
Comment (NabbeunNabi) [inline src/Admin/admin-api/admin-layout.tsx:53]: ♻️

File: src/App.tsx [hunk 21]
```diff
@@ -21,65 +21,76 @@ import MushroomGame from "./pages/MushroomGame";
...
 const AdminProtectedRoute = ({ children }: { children: React.ReactNode }) => {
-  const { user } = useAuth(); 
+ const { user } = useAuth();
 
-  if (!user) {  //redirect to login if user not authenticated
-    return <Navigate to="/login" />;
-  }
-  return <>{children}</>; //allow access if authenticated
+ if (!user) {
+  //redirect to login if user not authenticated
```
Comment (NabbeunNabi) [inline src/App.tsx:27]: 💯 great comment

File: src/App.tsx [hunk 22]
```diff
@@ -21,65 +21,76 @@ import MushroomGame from "./pages/MushroomGame";
...
+ useEffect(() => {
+  const lang = sessionStorage.getItem("lang");
+  if (lang) {
+   setCurrentLang(lang);
+   LanguageContext = createContext(lang);
+  } else {
+   setCurrentLang("jp");
+   LanguageContext = createContext("jp");
+  }
+ }, []);
```
Comment (NabbeunNabi) [inline src/App.tsx]: The comment says local storage but you are using session storage. Actually I think in your use case local storage would be great. Usually the user of the computer would want it to be the same always for the site. And not have to set always on tab visit. And it isn't private data. Just a locale key so it is safe to put in local storage.

File: src/pages/Home.tsx [hunk 23]
```diff
@@ -2,49 +2,45 @@ import Blogs from "../components/homeComponents/Blogs";
...
-      <Slogan></Slogan>
-      <MissionStatement/>
-      {/* <section id="cards" className={"min-h-80 flex flex-col px-4"}>
+ return (
+  <div className="bg-gradient-to-br from-green-300 to-white font-sans">
+   <div className="w-full m-auto"></div>
+   {images && <HeroCarousel images={images} />}
+   <Slogan></Slogan>
+   <MissionStatement />
+   {/* <section id="cards" className={"min-h-80 flex flex-col px-4"}>
```
Comment (NabbeunNabi) [inline src/pages/Home.tsx]: If not used in future ticket you can remove this. Always can add later

**Review by NabbeunNabi — CHANGES_REQUESTED** [review]

**Comment by NabbeunNabi** [comment]
@robertbolgeo reviewed! Looks much prettier 😁. Just a few things to address. Also I would update the title of the PR. It just says fix at the moment. You can add any title standard you want but it could follow conventional commits.

- Also your PR description says documentation is required but you haven't checked that it is updated. Just clarifying as to if it is done or if documentation is needed

File: lib/cjs/src/Admin/admin-api/admin-layout.d.ts (hunk 1 above)
Comment (robertbolgeo) [reply lib/cjs/src/Admin/admin-api/admin-layout.d.ts:7]: I've added a git issue for this problem #236

File: src/pages/Home.tsx (hunk 23 above)
Comment (NabbeunNabi) [reply src/pages/Home.tsx]: I see it was fixed in a different PR

File: src/App.tsx (hunk 22 above)
Comment (robertbolgeo) [reply src/App.tsx]: I changed it.

File: lib/cjs/src/components/homeComponents/Section.d.ts (hunk 13 above)
Comment (robertbolgeo) [reply lib/cjs/src/components/homeComponents/Section.d.ts]: This is caused by an empty component file being compiled at build. The component has been destroyed.

File: src/Admin/admin-api/admin-layout.tsx (hunk 15 above)
Comment (robertbolgeo) [reply src/Admin/admin-api/admin-layout.tsx:15]: made an issue to fix this #237

**Review by NabbeunNabi — APPROVED** [review]
Thanks for making those tickets and making the necessary changes

**Comment by robertbolgeo** [comment]
Added new issues for things I couldn't fix tonight, and fixed everything else. Thanks Will!

**Pull request merged**